torch/_inductor/utils.py

```diff
@@ -479,15 +490,61 @@ def get_kernel_metadata(node_schedule, wrapper):
             key = node.meta["from_node"][0][0]
             from_node_dict[key].append(node.name)
     metadata = (
-        f"{wrapper.comment} Source Nodes: [{', '.join(sorted(from_node_dict.keys()))}], "
-        f"Original ATen: [{', '.join(sorted(original_aten_dict.keys()))}]"
+        f"{wrapper.comment} Source Nodes: [{', '.join(from_node_dict.keys())}], "
+        f"Original ATen: [{', '.join(original_aten_dict.keys())}]"
     )
+
     # trace back to original node here
-    detailed_metadata = []
+    detailed_metadata = [f"{wrapper.comment} Source node to ATen node mapping:"]
     for original_node, nodes in sorted(from_node_dict.items()):
         detailed_metadata.append(
-            f"{wrapper.comment} {original_node} => {', '.join(sorted(nodes))}"
+            f"{wrapper.comment}   {original_node} => {', '.join(sorted(nodes))}"
         )
+
+    # print the aot_autograd graph fragment
+    if graph is not None:
+        detailed_metadata.append(f"{wrapper.comment} Graph fragment:")
+        node_names = {node.name for node in inductor_nodes}
+        # A bit hacky: get the parent graph generated code and only keep the
+        # lines corresponding to the variables we care about. This allows us
+        # to reuse the Python codegen code.
+        for line in graph.python_code("").src.split("\n"):
```

Comment:
Uhhh, this is a Graph, not a GraphModule, right? So the Python code is not cached, you're rendering it over and over again every time you print the aot autograd graph fragment. The Python code rendering is pretty expensive, definitely not recommended.